src/tsickle.ts

```diff
@@ -105,89 +105,173 @@ class ClosureRewriter extends Rewriter {
+        // list, as some information is only available in the former.
+        for (let i = 0, l = sig.parameters.length; i < l; i++) {
+          let paramNode = fnDecl.parameters[i];
+          let paramSym = sig.parameters[i];
+          let type = typeChecker.getTypeOfSymbolAtLocation(paramSym, fnDecl);
+
+          let newTag: jsdoc.Tag = {
+            tagName: 'param',
+            optional: paramNode.initializer !== undefined || paramNode.questionToken !== undefined,
+            parameterName: makeLegalExternsParam(unescapeName(paramSym.getName())),
+          };
+
+          let destructuring =
+              (paramNode.name.kind === ts.SyntaxKind.ArrayBindingPattern ||
+               paramNode.name.kind === ts.SyntaxKind.ObjectBindingPattern);
+
+          if (paramNode.dotDotDotToken !== undefined) {
+            newTag.restParam = true;
+            // In TypeScript you write "...x: number[]", but in Closure
+            // you don't write the array: "@param {...number} x".  Unwrap
+            // the Array<> wrapper.
+            type = (type as ts.TypeReference).typeArguments[0];
+          }
 
-        newTag.type = this.typeToClosure(fnDecl, type, destructuring);
+          newTag.type = this.typeToClosure(fnDecl, type, destructuring);
 
-        // Search for this parameter in the JSDoc @params.
-        for (let {tagName, parameterName, text} of jsDoc) {
-          if (tagName === 'param' && parameterName === newTag.parameterName) {
-            newTag.text = text;
-            break;
+          if (!isOverloaded) {
+            // Search for this parameter in the JSDoc @params.
+            // TODO: Consider adding text from existing JSDoc into overloads
+            for (let {tagName, parameterName, text} of jsDoc) {
+              if (tagName === 'param' && parameterName === newTag.parameterName) {
+                newTag.text = text;
+                break;
+              }
+            }
+            // push to the newDoc for rendering the single function
+            newDoc.push(newTag);
+          } else {
+            // either add an entry at this index, or update the existing name and/or type as needed
+            // merge parameter names at this index if required
+            if (newTag.parameterName) {
+              if (!paramData.names[i]) {
+                paramData.names[i] = newTag.parameterName;
+              } else if (paramData.names[i].indexOf(newTag.parameterName) === -1) {
+                paramData.names[i] += '_or_' + newTag.parameterName;
+              }
+              if (i >= minCount && paramData.names[i].indexOf('opt_') !== 0) {
+                paramData.names[i] = 'opt_' + paramData.names[i];
+              }
+            }
+            // merget type names at this index if required
+            if (newTag.type) {
+              if (!paramData.types[i]) {
+                paramData.types[i] = newTag.type;
+              } else if (paramData.types[i].indexOf(newTag.type) === -1) {
```

Comment:
Just build the list here, and join in the last step. This check would skip types if the later one was contained in an earlier: 
```
function(a: Array<SomeType>)
function(b: SomeType
```
would end up with only {Array<SomeType>}